src/components/activities/RosterPanel.tsx

```diff
@@ -82,13 +116,17 @@ export function ParticipantDialog({ open, participant, activity, onClose }: { op
             <Typography>Timeline:</Typography>
             <Table size="small">
               <TableBody>
-                {[...participant.timeline].reverse().map((t) => (
-                  <TableRow key={t.time}>
-                    <TableCell>{activity.organizations[t.organizationId].rosterName ?? activity.organizations[t.organizationId].title}</TableCell>
-                    <TableCell>{getStatusText(t.status)}</TableCell>
-                    <TableCell>{formatDate(t.time, 'EEE yyyy-MM-dd HHmm')}</TableCell>
-                  </TableRow>
-                ))}
+                {[...participant.timeline]
+                  .map((t, i) => (
+                    <TableRow key={t.time}>
+                      <TableCell>{activity.organizations[t.organizationId].rosterName ?? activity.organizations[t.organizationId].title}</TableCell>
+                      <TableCell>{getStatusText(t.status)}</TableCell>
+                      <TableCell>
+                        <EditTime datetime={t.time} onChange={(time) => updateTimeline({ ...t, time }, i)} />
+                      </TableCell>
+                    </TableRow>
+                  ))
+                  .reverse()}
```

Comment:
I put it after because the index is the identifier and needs to be set in the order that they appear in the original timeline array, even though we are displaying them in reverse order, here.

Reply:
Ahh makes sense! Works for me.